cross_list_courses/static/cross_list_courses/js/controllers/ListController.js

```diff
@@ -0,0 +1,107 @@
+                                '-'+course_instance.term.display_name+
+                                '-'+course_instance.course_instance_id;
+
+            };
+
+            $scope.dtOptions = {
+                ajax: function(data, callback, settings) {
+                    var url = $djangoUrl.reverse('icommons_rest_api_proxy',
+                                                   ['api/course/v2/xlist_maps/']);
+
+                    var queryParams = {
+                        offset: data.start,
+                        limit: data.length,
+                        include: 'course_instance',
+                    };
+
+                    $.ajax({
+                        url: url,
+                        method: 'GET',
+                        data: queryParams,
+                        dataSrc: 'data',
+                        dataType: 'json',
+                        success: function(data, textStatus, jqXHR) {
+                            callback({
+                                recordsTotal: data.count,
+                                recordsFiltered: data.count,
+                                data: data.results,
+                            });
+                        },
+                        error: function(data, textStatus, errorThrown) {
+                            $log.error('Error getting data from ' + url + ': '
+                                       + textStatus + ', ' + errorThrown);
+                            callback({
+                                recordsTotal: 0,
+                                recordsFiltered: 0,
+                                data: [],
+                            });
+                        },
+                    });
+                },
+                createdRow: function( row, data, dataIndex ) {
+                    // to use angular directives within the rendered datatable,
+                    // we have to compile those elements ourselves.  joy.
+                    $compile(angular.element(row).contents())($scope);
+                },
+                language: {
+                    emptyTable: 'There are no cross listed courses  to display.',
+                    info: 'Showing _START_ to _END_ of _TOTAL_ course mappings',
+                    infoEmpty: 'Showing 0 to 0 of 0 course mappings',
+                    paginate: {
+                        next: '',
+                        previous: '',
+                    },
+                },
+                lengthMenu: [10, 25, 50, 100],
+                // yes, this is a deprecated param.  yes, it's still required.
+                // see https://datatables.net/forums/discussion/27287/using-an-ajax-custom-get-function-don-t-forget-to-set-sajaxdataprop
+                sAjaxDataProp: 'data',
+                searching: false,
+                serverSide: true,
+            };
+
+
+            $scope.dtColumns = [
+                {
+                    data: null,
```

Comment:
I see what you meant in our stand-up discussion offline about duplication of code here. It would be nice if these could be in a modular function, yes. We can try to bang it out or let it go and mark it as a 'todo' for exploring as refactoring next time we're doing a story using this controller code.